apps/worker/src/app/workflow/usecases/send-message/send-message-chat.usecase.ts

```diff
@@ -167,33 +177,50 @@ export class SendMessageChat extends SendMessageBase {
       filterData: {
         tenant: command.job.tenant,
       },
-    });
+    };
 
-    if (!integration) {
-      await this.executionLogRoute.execute(
-        ExecutionLogRouteCommand.create({
-          ...ExecutionLogRouteCommand.getDetailsFromJob(command.job),
-          detail: DetailEnum.SUBSCRIBER_NO_ACTIVE_INTEGRATION,
-          source: ExecutionDetailsSourceEnum.INTERNAL,
-          status: ExecutionDetailsStatusEnum.FAILED,
-          isTest: false,
-          isRetry: false,
-          raw: JSON.stringify({
-            reason: `Integration with integrationId: ${subscriberChannel?._integrationId} is either deleted or not active`,
-          }),
-        })
-      );
+    /**
+     * Current a workaround as chat providers for whatsapp is more similar to sms than to our chat implementation
+     */
+    if (subscriberChannel.providerId === ChatProviderIdEnum.WhatsAppBusiness) {
+      delete searchCriteria.id;
+    }
 
+    const integration = await this.getIntegration(searchCriteria);
+
+    if (subscriberChannel.providerId !== ChatProviderIdEnum.WhatsAppBusiness) {
+      if (!integration) {
+        await this.executionLogRoute.execute(
+          ExecutionLogRouteCommand.create({
+            ...ExecutionLogRouteCommand.getDetailsFromJob(command.job),
+            detail: DetailEnum.SUBSCRIBER_NO_ACTIVE_INTEGRATION,
+            source: ExecutionDetailsSourceEnum.INTERNAL,
+            status: ExecutionDetailsStatusEnum.FAILED,
+            isTest: false,
+            isRetry: false,
+            raw: JSON.stringify({
+              reason: `Integration with integrationId: ${subscriberChannel?._integrationId} is either deleted or not active`,
+            }),
+          })
+        );
+
+        return;
+      }
+    } else if (!integration) {
+      /**
+       * TODO: Need to handle a proper execution log error for this case
+       */
       return;
     }
 
     const chimeraOverride = this.getChimeraOverride(command.chimeraData?.providers, integration);
 
     const chatWebhookUrl =
       chimeraOverride?.webhookUrl || command.payload.webhookUrl || subscriberChannel.credentials?.webhookUrl;
+    const phoneNumber = subscriberChannel.credentials?.phoneNumber;
     const channelSpecification = subscriberChannel.credentials?.channel;
 
-    if (!chatWebhookUrl) {
+    if (!chatWebhookUrl && integration.providerId !== ChatProviderIdEnum.WhatsAppBusiness) {
```

Comment:
Those workaround will be handeled during a more thorough refactoring of the "Chat" Channel